test/test_parse_array.py

```diff
@@ -119,3 +132,93 @@ def test_transform_layered_array():
     assert result["data"].shape == (2, 8)
     assert result["data"].dims == ("layer", "dim_0")
     assert np.array_equal(result["data"][0], np.ones((8,)))
+
+
+def test_transform_full_component():
+    grammar = """
+start: block*
+block: options_block | arrays_block
+options_block: "begin"i "options"i options_vars "end"i "options"i
```

Comment:
grammars like these, aware of the names/contents of blocks, and the names/types of variables, can be generated for each component